bindings/c/test/fdb_api.hpp

```diff
@@ -147,6 +187,52 @@ struct KeyValueArray {
 		return Error(err);
 	}
 };
+struct KeyValueArray {
+	using Type = std::pair<std::vector<KeyValue>, bool>;
+	static Error extract(native::FDBFuture* f, Type& out) noexcept {
```

Comment:
This is a thinker for me.
I want this header to be performance-light as possible, preserve the existing contract of the C API (so that I don't have to make new decisions here every time API changes), and do as little on top of C API as possible. Heap allocation + linear copy feels like a surprise, a crossing of some metaphorical line. I'd also like to avoid `Native` prefix being visible in the `fdb` namespace.
Let's add something like below and keep the original `KeyValueArray` (and do the same for `KeyRange`):
```
struct KeyValue : native::FDBKeyValue {
    KeyRef key() const noexcept { return KeyRef(key, intSize(key_length));
    ValueRef value() const noexcept { return ValueRef(value, intSize(value_length)); }
};
...
struct KeyValueArray {
    using Type = std::tuple<KeyValue const*, int, bool>;
...
```

Reply:
Done, but I kept Ref in the name to be consistent with the input interface, and to make it very clear that the return value is a reference and won't survive after future is destructed.